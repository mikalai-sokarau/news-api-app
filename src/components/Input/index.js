import React from 'react';

const Input = ({ onChangeHandler, placeholder }) => (
  <div className="input-field">
    <input
      onChange={({ target: { value } }) => onChangeHandler(value)}
      placeholder={placeholder}
      type="text"
      className="validate"
    />
  </div>
);

export default Input;
