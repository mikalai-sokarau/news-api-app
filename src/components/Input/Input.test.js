import React from 'react';
import { shallow, mount } from 'enzyme';
import Input from '../Input';

describe('<Input />', () => {
  test('render an Input', () => {
    const wrapper = shallow(<Input />);
    
    expect(wrapper).toMatchSnapshot();
  });

  test('render props correctly', () => {
    const wrapper = shallow(<Input placeholder="hello" />);

    expect(wrapper.children().props().placeholder).toBe('hello');
  });

  test('callback works correctly', () => {
    const callback = jest.fn();
    const wrapper = shallow(<Input onChangeHandler={callback} />);

    expect(wrapper).toMatchSnapshot();
    wrapper.find('input').simulate('change', { target: { value: 'value' } });
    expect(callback).toBeCalledWith('value');
  });
});
