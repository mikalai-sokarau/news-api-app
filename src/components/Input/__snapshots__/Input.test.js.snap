// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> callback works correctly 1`] = `
<div
  className="input-field"
>
  <input
    className="validate"
    onChange={[Function]}
    placeholder=""
    type="text"
  />
</div>
`;

exports[`<Input /> render an Input 1`] = `
<div
  className="input-field"
>
  <input
    className="validate"
    onChange={[Function]}
    placeholder=""
    type="text"
  />
</div>
`;
